bigtable/client/internal/row_key_compare.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/internal/prefix_range_end.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace internal {
+
+int RowKeyCompare(absl::string_view lhs, absl::string_view rhs) {
```

Comment:
Can we reuse [`absl::string_view::compare()`](https://github.com/abseil/abseil-cpp/blob/0271cd35577599fa99b59202da17d3136956e4c0/absl/strings/string_view.h#L351-L369) here? It feels like this is a reimplementation of that, unless we have something special here.

Reply:
I think the implementation of `absl::string_view::compare()` is incorrect.  I understand `absl::string_view` is supposed to implement the C++17 `std::string_view`, which should compare using the character traits (cf page 772 of the [C++17 draft](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4659.pdf) or the [online doc](http://en.cppreference.com/w/cpp/string/basic_string_view/compare)).

For platforms where `char` is signed, then `memcmp()` and `traits::compare()` yield different results if the characters have values in the `\x80` to `\xFF` range.

Reply:
Well, I learned something cool today.  `std::char_traits<char>` specializes the `lt` comparison to compare as `unsigned char` values.  And thus `std::memcmp` is the right thing in `string_view` (and `string`).

I swear I got a test failure because of signedness, but it must have been just a bad test that got me confused.

Reply:
Removed the file.

Reply:
Thanks for verifying this! Less code is better. 😄